### PR TITLE
feat: add seo/twitter component for twitter cards, og tags, etc.

### LINKED ISSUE #4: Make Twitter cards work
Seems like there's no fancy preview

### PR DESCRIPTION
**Pull request by DSchau**
Fixes #4 

This PR adds an (automatic, mostly) twitter card capability, as well as some other general SEO goodness.

There's more to do here, but I didn't want to scope creep. Specifically, I think the following are worth doing:

- Move posts to content/ (better fit than pages), and cleans up directory structure a bit
- Add images/featured images to post, and inject them with GraphQL, pass to the SEO component (this will improve the twitter cards)
  - This is set up, just didn't add any images yet

### CODE REVIEW

File: src/components/SEO.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        siteUrl
+        social {
+          twitter
+        }
+      }
+    }
+  }
+`
+
+function SEO({ meta, image, title, description }) {
```
Comment (DSchau) [inline src/components/SEO.js]: See [this guide](https://www.gatsbyjs.org/docs/add-seo-component/) for more info. Reusability here is pretty great, and it's nice to default some fields.

File: src/pages/index.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import React from 'react'
 import { Link, graphql } from 'gatsby'
 import get from 'lodash/get'
-import Helmet from 'react-helmet'
 
 import Bio from '../components/Bio'
 import Layout from '../components/Layout'
+import SEO from '../components/SEO'
```
Comment (DSchau) [inline src/pages/index.js]: This should be imported into _every_ component that will be served to the end user. It's currently just the index and blog post template, but additional pages are added (e.g. about, contact, etc.) definitely want to re-use this.

File: src/templates/blog-post.js [hunk 3]
```diff
@@ -1,27 +1,22 @@
...
     const { previous, next } = this.props.pageContext
 
     return (
       <Layout location={this.props.location} title={siteTitle}>
-        <Helmet
-          htmlAttributes={{ lang: 'en' }}
-          meta={[{ name: 'description', content: siteDescription }]}
-          title={`${post.frontmatter.title} | ${siteTitle}`}
-        />
+        <SEO title={post.frontmatter.title} description={post.excerpt} />
```
Comment (DSchau) [inline src/templates/blog-post.js]: You'll generally _always_ want to pass title. description defaults to siteMetadata.description.

Longer term, image can be passed as well, which will be handy for SEO, twitter cards, etc.

**Comment by DSchau** [comment]
An example of how it looks when shared via Twitter.

screen shot 2018-12-04 at 11 06 44 am

See the [staged site here](https://5c06b65f41174f1038d41b72--gracious-euclid-09716b.netlify.com/)

File: src/components/SEO.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        const { siteMetadata } = data.site
+        const metaDescription = description || siteMetadata.description
+        const metaImage = image ? `${siteMetadata.siteUrl}/${image}` : null
+        const url = `${siteMetadata.siteUrl}${slug}`
+        return (
+          <Helmet
+            htmlAttributes={{ lang: 'en' }}
+            {...(title
+              ? {
+                  titleTemplate: `%s - ${siteMetadata.title}`,
```
Comment (DSchau) [inline src/components/SEO.js]: react-helmet has some weird behavior here. If you _don't_ pass a title (e.g. for the index route, we want the title to just be the site name) it renders empty/null.

So we need a check here to only use titleTemplate if we have a title passed in.

**Comment by gaearon** [comment]
thanks

**Comment by gaearon** [comment]
@DSchau Do you mind also fixing twitter card for main page?

**Comment by DSchau** [comment]
@gaearon sure—I’ll fix tomorrow!

**Comment by gaearon** [comment]
nvm, I figured it out :-)

**Pull request merged**